chromium/rules.js

```diff
@@ -15,6 +15,15 @@ const trivial_rule_from_c = new RegExp("^http:");
 const trivial_cookie_name_c = new RegExp(".*");
 const trivial_cookie_host_c = new RegExp(".*");
 
+// Empty iterable singleton to reduce memory usage (#12494).
+const nullIterable = Object.create(null, {
+  [Symbol.iterator]: {
+    value: function* () {
+      // do nothing
```

Comment:
This can be done in a simpler way.

```js
const nullIterator = {
  next() {
    return { done: true }
  }
}

const nullIterable = {
  [Symbol.iterator]: function () {
    return nullIterator
  }
}
```

Will also save memory.

Reply:
Having just one name is nice. How about just:
```javascript
const nullIterable = Object.create(null, {
  [Symbol.iterator]: {
    value: () => ({
      next() {
        return {done: true};
      }
    })
  }
});

```

Reply:
@cowlicks `const nullIterable = (function* () {})()` would be better.

Reply:
@cowlicks why use `Object.create(null, { ...`?

Reply:
@cowlicks This will create a new instance of `{ next() { return {done: true}; } }` each time.

Reply:
Currently I have this code:
```js
const nullIterator = {
  next() {
    return { done: true };
  }
};

const nullIterable = {
  [Symbol.iterator]: function () {
    return nullIterator;
  },
  size: 0
};
```

Reply:
@koops76 see https://github.com/EFForg/https-everywhere/pull/12577#discussion_r139248205

re: new instance of ... thing - that is fine, it should get GC'd immediately anyway